From 6759cf3d83e2e362166641dc2747c35ddaf4e733 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Thu, 31 Oct 2013 17:11:29 +0800
Subject: [PATCH 39/70] ENGR00285651 Subtitle:Some redundant characters were
 shown on the screen

remove tags for subtitle location before displaying it.

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
index c6f25aa..73e73f4 100755
--- a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
+++ b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
@@ -624,7 +624,11 @@ public class VideoPlayer extends Activity implements SeekBar.OnSeekBarChangeList
             if(mPlaySpeed == 1){
                 if(text != null){
                     //Log.d(TAG,"text is " + text.getText());
-                    mSubtitleTextView.setText(text.getText(), 0);
+                    int start_pos = 0;
+                    if(text.getText().startsWith("{\\pos")){
+                        start_pos = text.getText().indexOf('}') + 1;
+                    }
+                    mSubtitleTextView.setText(text.getText().substring(start_pos), 0);
                 }
                 else
                     mSubtitleTextView.setText(null, 0);
-- 
1.8.0

